---
title: "Prueba palmscan"
author: "Sergio Diez Hermano"
date: '`r format(Sys.Date(),"%e de %B, %Y")`'
output:
  html_document:
      highlight: tango
      code_folding: show
      toc: yes
      toc_depth: 4
      toc_float:
        collapsed: yes
editor_options: 
  markdown: 
    wrap: sentence
---

```{r setup, include=FALSE}
require(knitr)
# include this code chunk as-is to set options
opts_chunk$set(comment = NA, prompt = FALSE, fig.height = 5, fig.width = 5, dpi = 300, fig.align = "center", 
               message = FALSE, warning = FALSE, cache = FALSE, verbose = FALSE)
# Sys.setenv(RSTUDIO_PANDOC = "C:/Program Files/Pandoc")
Sys.setlocale("LC_TIME", "C")
```

# Libraries

```{r}
library(stringr)    # extract strings
library(data.table) # read tables
library(seqinr)     # read fasta files
library(ggplot2)    # Plotting
library(viridis)    # Plotting
library(RColorBrewer)
library(ggtree)
library(ggmsa)
library(DescTools)  # capitalise first letter
library(ggseqlogo)

```

# Load and format data

```{r}
# Load palmprints
palmprints <- fread("palmscan_palmprints/pp.fev", header = F)
colnames(palmprints) <- tstrsplit(palmprints[1,], "=")[1] %>% unlist()
palmclean <- apply(palmprints, 1, function(x) gsub(".*=","", x)) %>% t() %>% as.data.table()

# Load diamond matches with PALMdb
palmdb <- fread("diamond_outputs/palmprint_search_diamond.out")
colnames(palmdb) <- c("query_name", "query_start_pos", "query_end_pos", "query_total_length", 
                      "target_name", "target_start_pos", "target_end_pos", "target_total_length", 
                      "percent_identity", "expectance_value", "CIGAR", "target_sequence")

# Load taxa database
palmtaxa <- fread("C:/Users/Usuario/MSI Stealth/Yutera/jcyl_viromics/databases/palmdb-main/2021-03-14/u_tax.tsv")

# Load fasta per phylym
msafiles <- list.files("msa/mafft_einsi_perphylum/", pattern = ".fasta", full.names = T)
msaphylum <- sapply(msafiles, function(x) read.fasta(x, seqtype = "AA"))

```

# Check PALMdb taxonomy assignment

```{r}
# List of unique contig names. 146 found in PALMdb, 12 missing
uniqvir <- unique(palmdb$query_name)

# Function to extract taxonomy
extract_tax <- function(contig, diamond, ids, pi) {
  
  uvir1 <- diamond[query_name == contig, ]
  uvir2 <- ids[ids$Label %in% uvir1$target_name, ]
  uvir3 <- uvir2[order(match(uvir2$Label, uvir1$target_name)), ]
  uvir4 <- cbind(uvir3, uvir1[, c("percent_identity", "expectance_value")])
  uvir5 <- uvir4[percent_identity >= pi, ]
  uvir6 <- uvir5[order(percent_identity, decreasing = T), ]
  return(uvir6)
  
}

# Extract top 10 taxonomy and store in list
uvir.palmdb <- lapply(uniqvir, function(x) extract_tax(x, palmdb, palmtaxa, 30)[1:10, ] %>% na.omit)
names(uvir.palmdb) <- uniqvir

# # Select most detailed taxonomy from top 10
# uvir.palmdb.empty <- lapply(uvir.palmdb, 
#                             function(x) apply(x, 1, 
#                                               function(y) length(grep("\\.", y[2:7]))))
# uvir.palmdb.pos <- sapply(uvir.palmdb.empty, function(x) which.min(x))
# uvir.palmdb.chosen <- lapply(1:length(uniqvir), function(x) uvir.palmdb[[x]][uvir.palmdb.pos[[x]]])

# Select taxonomy with best percent identity
uvir.palmdb.chosen <- lapply(uvir.palmdb, function(x) x[1, ])

# As data frame
uvir.palm.frame <- cbind(contig = uniqvir, do.call(rbind, uvir.palmdb.chosen))
uvir.palm.frame$source <- rep("palmdb", dim(uvir.palm.frame)[1])

```

# Try to fill blanks with RdRpscan

## Format RdRpscan table with taxa

```{r}
# Load diamond matches with RdRp-scan
rdrpscan <- fread("diamond_outputs/rdrpscan_search_diamond.out")
colnames(rdrpscan) <- c("query_name", "query_start_pos", "query_end_pos", "query_total_length", 
                      "target_name", "target_start_pos", "target_end_pos", "target_total_length", 
                      "percent_identity", "expectance_value", "CIGAR", "target_full_sequence", "target_align_fragment")

# Format rdrpscan species
rdrpscan$species <- sapply(rdrpscan$target_name, function(x) ifelse(grepl("\\[", x), 
                                             str_extract_all(x, "(?<=\\[).+?(?=\\])")[[1]],
                                             x
                                             )) %>% gsub("_", " ", .)

# List of unique contig names. 146 found in PALMdb, 12 missing
uniqvir.scan <- unique(rdrpscan$query_name)

# Extract taxonomy and store in list
uvir.scan <- lapply(uniqvir.scan, function(x) rdrpscan[query_name == x, ][order(expectance_value), ][1,])

# As data frame
uvir.scan.frame <- do.call(rbind, uvir.scan)

# Retrieve taxonomy from NCBI
# Do only once, then load RData
# rdrpscan.ncbitax <- sapply(uvir.scan.frame$species, function(x) myTAI::taxonomy(organism = x, 
#                                                                                 db       = "ncbi",
#                                                                                 output   = "classification" ))
# saveRDS(rdrpscan.ncbitax, "rdrpscan_ncbitax.RData")
rdrpscan.ncbitax <- readRDS("rdrpscan_ncbitax.RData")

# Extract taxonomy info from NCBI list
getncbitax <- function(ncbidat) {
  
  if (is.na(ncbidat)[1]) {
    
    taxclas <- rep(NA, 6) %>% t %>% as.data.frame
    colnames(taxclas) <- c("phylum", "class", "order", "family", "genus", "species")
    
  } else {
    
    taxtras <- ncbidat %>% t
    colnames(taxtras) <- taxtras[2, ]
    taxtras <- taxtras[1, ]
    taxclas <- taxtras[names(taxtras) %in% c("phylum", "class", "order", "family", "genus", "species")] %>% t %>% as.data.frame()
    
  }
  
  return(taxclas)
  
}

# Apply extractor function
rdrpscan.ncbiget <- lapply(rdrpscan.ncbitax, function(x) getncbitax(x))
# Reorder taxonomy columns
rdrpscan.ncbiframe <- rbindlist(rdrpscan.ncbiget, fill = T)[, c(1,2,3,4,6,5)]
# Add original species names by Rdrpscan
rdrpscan.ncbiframe$species <- uvir.scan.frame$species
# Format similar to palmdb data frame
rdrpscan.ncbiframe <- cbind(contig = uvir.scan.frame$query_name,
                            rdrpscan.ncbiframe, 
                            uvir.scan.frame[, c("percent_identity", "expectance_value")])
rdrpscan.ncbiframe$source <- rep("rdrpscan", dim(rdrpscan.ncbiframe)[1])


```

## Fill in blanks in uvir.palm.frame

```{r}
# Get empty entries in palmdb taxa
palmdb.empty.pos <- which(apply(uvir.palm.frame[,3:8], 
                                1, 
                                function(x) paste(x, collapse = "")) == "......")
# Fill in with rdrpscan taxa
mixed.frame <- uvir.palm.frame
mixed.frame[palmdb.empty.pos, 3:11] <- rdrpscan.ncbiframe[palmdb.empty.pos, 2:10]

# Fill in taxonomy for "-like" virus
mixed.frame$species <- gsub("-like", "", mixed.frame$species)
# Get index of "like" viruses and position of "like" term in their names
viruslike <- grep("like", mixed.frame$species)
likepos <- lapply(str_split(mixed.frame$species, " "), function(x) grep("like", x)) %>% unlist
# Get the "like" particle within the list of "like" viruses and delete it
likeparticles <- lapply(1:length(likepos), 
                        function(x) str_split(mixed.frame$species[viruslike[x]], " ")[[1]][likepos[x]]) %>% 
  gsub("like", "", .) %>%
  gsub("botouli", "botourmia", .) %>%
  str_to_title()
# Find row and column match of first apparition of the "like" virus in other taxonomy entry
rowlike <- sapply(likeparticles, function(x) grep(x, apply(mixed.frame[, 3:7], 1, function(x) paste(x, collapse= "")))[1])
collike <- sapply(1:length(rowlike), 
                  function(x) grep(likeparticles[x], mixed.frame[rowlike[x], 3:7]) + 2) %>% unlist
# Replace with most similar taxonomy
for (i in 1:length(viruslike)) {
  
  mixed.frame[viruslike[i], 3:collike[i]] <- mixed.frame[rowlike[i], 3:collike[i]]
  
}

# Replace NA by "."
mixed.frame[is.na(mixed.frame)] <- "."
write.csv(mixed.frame, "palmdb_rdrpscan_frame.csv")

# If species PI is bellow 90%, discard species
mixed.frame[mixed.frame$percent_identity < 90, "species"] <- "."
write.csv(mixed.frame, "palmdb_rdrpscan_frame_species90thres.csv")
```


# Create meta data

FP: Pinus radiata, CC: Castanea sativa, PQ: Quercus ilex
1: tree, 2: fungi from tree, 3: fungi from insects, 4: insects

```{r}
# Get abbreviation from contig name
contigname <- sapply(mixed.frame$contig, function(x) str_split(x, "_")[[1]][1]) %>% unlist

# Short names
treeshort <-  gsub('FP', 'Prad',
                     gsub('CC', 'Csat',
                          gsub('PQ', 'Qile', contigname)))
metashort <- gsub('1', '_Tree',
                  gsub('2', '_TreeFungi',
                       gsub('3', '_ArthFungi', 
                            gsub('4', '_Arth', treeshort))))

# Long names
treelong <- contigname
treelong[grep("FP", contigname)] <- "Pinus radiata"
treelong[grep("CC", contigname)] <- "Castanea sativa"
treelong[grep("PQ", contigname)] <- "Quercus ilex"

samplelong <- contigname
samplelong[grep("1", contigname)] <- "Tree"
samplelong[grep("2", contigname)] <- "Fungi from tree"
samplelong[grep("3", contigname)] <- "Fungi from arthropod"
samplelong[grep("4", contigname)] <- "Arthropod"

# # Add known taxonomy (excluding species name for no PI threshold)
# repres.taxa <- apply(mixed.frame[,c(2:7)], 1, 
#                      function(x) ifelse(x[6] == ".", 
#                                         x[which(x == ".") - 1],
#                                         x[6]))
# repres.label <- apply(mixed.frame[,c(2:7)], 1, 
#                      function(x) ifelse(x[6] == ".", 
#                                         names(x[which(x == ".") - 1]),
#                                         names(x[6]))) %>% StrCap(., "first")
# repres.label[repres.label == "Label"] <- "NA"

# Add known taxonomy (including species name for PI threshold)
repres.taxa <- apply(mixed.frame[,c(2:8)], 1, 
                     function(x) ifelse(x[7] == ".", 
                                        x[which(x == ".")[1] - 1],
                                        x[7]))
repres.label <- apply(mixed.frame[,c(2:8)], 1, 
                     function(x) ifelse(x[7] == ".", 
                                        names(x[which(x == ".")[1] - 1]),
                                        names(x[7]))) %>% StrCap(., "first")
repres.label[repres.label == "Label"] <- "NA"

# Meta data frame
meta.frame <- data.frame(contig = names(metashort), 
                         short = metashort, 
                         tree = treelong, 
                         sample = samplelong,
                         repres_taxa = repres.taxa,
                         short_taxa = paste0(metashort, "_", repres.taxa),
                         repres = repres.label,
                         repres_label = paste0(repres.label, "_", repres.taxa),
                         row.names = NULL)
write.csv(meta.frame, "metadata_species90thres.csv")

```

# Figures

## Load data

```{r}
# Load mixed.frame and meta.frame
mixed.frame <- fread("palmdb_rdrpscan_frame_species90thres.csv", drop=1)
meta.frame <- fread("metadata_species90thres.csv", drop=1, header = T)

# Remove bacteria and fungal (Lasius) contigs
mixed.frame <- mixed.frame[-grep("bacter", mixed.frame$genus), ]
mixed.frame <- mixed.frame[-grep("Lasius", mixed.frame$genus), ]

meta.frame <- meta.frame[-grep("bacter", meta.frame$repres_taxa), ]
meta.frame <- meta.frame[-grep("Lasius", meta.frame$repres_taxa), ]

```

### Add info about unknown sequences

```{r}
# Load palmprints
palmprints <- fread("palmscan_palmprints/pp.fev", header = F)
colnames(palmprints) <- tstrsplit(palmprints[1,], "=")[1] %>% unlist()
palmclean <- apply(palmprints, 1, function(x) gsub(".*=","", x)) %>% t() %>% as.data.table()

# # Get all unknown contigs
# unkcontig <- palmclean$query[-which(palmclean$query %in% mixed.frame$contig)]
# Get unknown contigs that didn't have matches in blastp
unkcontig <- c("PQ2_NODE_10979_length_1996_cov_7.508580", "PQ3_NODE_2108_length_3144_cov_5.901009")

# Get abbreviation from contig name
contigname <- sapply(unkcontig, function(x) str_split(x, "_")[[1]][1]) %>% unlist

# Short names
treeshort <-  gsub('FP', 'Prad',
                     gsub('CC', 'Csat',
                          gsub('PQ', 'Qile', contigname)))
metashort <- gsub('1', '_Tree',
                  gsub('2', '_TreeFungi',
                       gsub('3', '_ArthFungi', 
                            gsub('4', '_Arth', treeshort))))

# Long names
treelong <- contigname
treelong[grep("FP", contigname)] <- "Pinus radiata"
treelong[grep("CC", contigname)] <- "Castanea sativa"
treelong[grep("PQ", contigname)] <- "Quercus ilex"

samplelong <- contigname
samplelong[grep("1", contigname)] <- "Tree"
samplelong[grep("2", contigname)] <- "Fungi from tree"
samplelong[grep("3", contigname)] <- "Fungi from arthropod"
samplelong[grep("4", contigname)] <- "Arthropod"

# Add known taxonomy (excluding species name)
repres.taxa <- rep("Unknown", length(unkcontig))

# Super meta data frame
unk.frame <- data.frame(contig = names(metashort), 
                        short = metashort, 
                        tree = treelong, 
                        sample = samplelong,
                        repres_taxa = repres.taxa,
                        short_taxa = paste0(metashort, "_", repres.taxa),
                        repres = rep("Unknown", length(unkcontig)),
                        repres_label = rep("Unknown", length(unkcontig)),
                        row.names = NULL)
meta.frame2 <- rbind(meta.frame, unk.frame)
meta.frame2$phyl_taxa <- paste0(meta.frame2$short_taxa, 
                                "_", 
                                c(mixed.frame$phylum, rep("unknown", length(unkcontig))))

# Super mixed data frame
unk.mixed <- rbind(c(unkcontig[1], "Unknown", rep(".", 6), c(0,0), "NA"),
                   c(unkcontig[2], "Unknown", rep(".", 6), c(0,0), "NA")) %>% as.data.frame
colnames(unk.mixed) <- colnames(mixed.frame)
mixed.frame2 <- rbind(mixed.frame, unk.mixed)

# Prepare taxa frame for plotting and working
working.frame <- mixed.frame2
working.frame[working.frame == "."] <- "Not assigned"
working.frame[working.frame$genus == "Botybirnavirus", "family"] <- "Botybirnavirus"
```

## Frequencies

```{r}
# Frequency table
freq.summary <- table(meta.frame2$sample, meta.frame2$tree)
colnames(freq.summary) <- c("C.sativa", "P.radiata", "Q.ilex")

# Plot
png(filename="figures/barplot.png", type="cairo",
    units="in", width=5, height=5, pointsize=12,
    res=300)

par(mar=c(2,4,1,1))
barplot(freq.summary, 
        col = colorRampPalette(brewer.pal(n = 8, name = "Dark2"))(12)[c(8,5,1,3)],
        border="white", 
        space=0.04,
        # legend = T,
        ylim = c(-2, 85),
        ylab = "Frequency")
legend("topleft", 
       legend = rownames(freq.summary) %>% rev, 
       fill = colorRampPalette(brewer.pal(n = 8, name = "Dark2"))(12)[c(3,1,5,8)],
       bty = "n",
       cex = 0.8)
box()

dev.off()

```

## Heatmap

```{r, fig.width=8}
# Frequency table for heatmap
freqfam <- table(meta.frame2$short, working.frame$family)
freqfam <- freqfam[, names(sort(table(working.frame$family), index.return = T, decreasing = T))]
freqfam <- freqfam[, -1]

# png(filename="heatmap.png", type="cairo",
#     units="in", width=5, height=5, pointsize=12,
#     res=300)
# heatmap(freqfam, Colv = NA, scale = "none", col = viridis(1000, direction = -1, option = "F"))
# dev.off()

# Column legend colors
ann_col <- unique(meta.frame2[, c("sample", "short")])
rownames(ann_col) <- ann_col$short
ann_col[, 2] <- NULL
ann_col$species <- rep(c("Chestnut", "Pinus", "Oak"), each = 4)
samplecols <- colorRampPalette(brewer.pal(n = 8, name = "Dark2"))(12)[c(1,3,5,8)]
names(samplecols) <- unique(meta.frame2$sample)

# Row legend colors
ann_row <- data.frame(genome = c("ssRNA(+)", "dsRNA", "ssRNA(+)", "ssRNA(+)", "dsRNA", "ssRNA(+)", 
                                 "dsRNA", "ssRNA(+)", "ssRNA(+)", "ssRNA(+)", "ssRNA(+)", "dsRNA", "ssRNA(-)", 
                                 "ssRNA(+)", "ssRNA(+)", "ssRNA(+)", "ssRNA(+)", "ssRNA(-)", "dsRNA"))
rownames(ann_row) <- colnames(freqfam)
genomes <- c("darkgrey", "black", "white")
names(genomes) <- unique(ann_row$genome)

# Colors
ann_color <- list(sample = samplecols,
                  species = c(Chestnut = "Orange", Pinus = "Purple", Oak = "Lightblue"),
                  genome = genomes)

png(filename="figures/heatmap.png", type="cairo",
    units="in", width=8, height=5, pointsize=12,
    res=300)
pheatmap::pheatmap(t(freqfam),
                   cluster_rows = T,
                   cluster_cols = T,
                   angle_col = 45,
                   annotation_col = ann_col,
                   annotation_row = ann_row,
                   annotation_colors = ann_color,
                   # show_colnames = F,
                   legend = T,
                   color = c("white", colorRampPalette(brewer.pal(n = 9, name = "RdPu"))(1000)[100:1000]))
dev.off()
```

## Tree

### Plot tree with taxa label

```{r, fig.width=20, fig.height=20}
## IQTREE
# tree <- read.tree("msa/mafft_einsi/iqtree/mafft_einsi_2021jcyl.fasta.contree")
tree <- read.tree("msa/mafft_einsi_superclean/iqtree/mafft_einsi_2021jcyl_superclean.fasta.contree")
tree$tip.label <- meta.frame2$repres_label[sapply(tree$tip.label, function(x) which(x == meta.frame2$contig))]

# # Convert MSA names into tree names for msaplot
# # msaiqtree <- read.fasta("msa/mafft_einsi/mafft_einsi_2021jcyl.fasta")
# msaiqtree <- read.fasta("msa/mafft_einsi_superclean/mafft_einsi_2021jcyl_superclean.fasta")
# names(msaiqtree) <- tree$tip.label
# write.fasta(sequences = msaiqtree, names = tree$tip.label,
#             "msa/mafft_einsi_superclean/mafft_einsi_2021jcyl_superclean_taxanames.fasta")

# Plot tree with MSA
addmeta <- cbind(meta.frame2$repres_label, meta.frame2[, -"repres_label"])
p <- ggtree(tree) %<+% addmeta +
  # geom_tiplab(aes(color = sample), align=TRUE) +
  geom_tiplab(align=TRUE, nudge_x = 0.25) +
  # geom_text(aes(label = tree$node.label), hjust = 1, vjust = -0.4, size = 3) +
  geom_nodelab(nudge_x = 0.035, size = 2.5) +
  # geom_nodelab(nudge_x = -0.125, nudge_y = 0.55, size = 2) +
  # geom_nodelab(nudge_x = 0.15, nudge_y = 0, size = 2) +
  theme_tree2() +
  xlim(0,10)

# Save tree with MSA
png(filename="figures/tree_MSA_taxa.png", type="cairo",
    units="in", width=18, height=20, pointsize=12,
    res=300)

msaplot(p=p, fasta="msa/mafft_einsi_superclean/mafft_einsi_2021jcyl_superclean_taxanames.fasta", 
        width = 0.75,
        offset = 2) +
  theme(legend.position = "none")

dev.off()

# CIRCULAR PLOTS AT END OF TIPS
# https://www.biostars.org/p/474821/
```

### Plot tree with shapes and colors

```{r, fig.width=20, fig.height=20}
## IQTREE
tree <- read.tree("msa/mafft_einsi_superclean/iqtree/mafft_einsi_2021jcyl_superclean.fasta.contree")

# Plot tree with MSA
p <- ggtree(tree) %<+% meta.frame2 +
  # geom_tiplab(aes(color = sample), align=TRUE) +
  geom_tiplab(align=TRUE, nudge_x = 0.25) +
  geom_tippoint(aes(shape = tree, colour = sample), size = 3, position = position_nudge(x = 0.1)) +
  scale_color_manual(values = colorRampPalette(brewer.pal(n = 8, name = "Dark2"))(12)[c(8,5,1,3)]) +
  # geom_text(aes(label = tree$node.label), hjust = 1, vjust = -0.4, size = 3) +
  geom_nodelab(nudge_x = 0.035, size = 2.5) +
  # geom_nodelab(nudge_x = -0.125, nudge_y = 0.55, size = 2) +
  # geom_nodelab(nudge_x = 0.15, nudge_y = 0, size = 2) +
  theme_tree2() +
  xlim(0,10)

# Save tree with MSA
png(filename="figures/tree_MSA_rightdots.png", type="cairo",
    units="in", width=18, height=20, pointsize=12,
    res=300)

msaplot(p=p, fasta="msa/mafft_einsi_superclean/mafft_einsi_2021jcyl_superclean.fasta", 
        width = 0.75,
        offset = 2) +
  theme(legend.position = "none")

dev.off()

# Save tree with tipshapes and tipcolors in one column 
png(filename="figures/tree_tipshapes_colors.png", type="cairo",
    units="in", width=15, height=20, pointsize=12,
    res=300)

ggtree(tree, branch.length = "none") %<+% meta.frame2 + 
  # geom_tiplab(align=TRUE) +
  geom_tippoint(aes(shape = tree, fill = sample), size = 3, position = position_nudge(x = 0.5)) +
  scale_shape_manual(values=c(21,24,22))
  # scale_shape(solid = F)

dev.off()
```

### Reorder msa according to tree

After running this chunk go to NCBI to plot MSA and save as SVG

```{r, fig.height=20}

tree <- read.tree("msa/mafft_einsi_superclean/iqtree/mafft_einsi_2021jcyl_superclean.fasta.contree")

h <- ggtree(tree) +
  # geom_tiplab(aes(color = sample), align=TRUE) +
  geom_tiplab(align=TRUE, nudge_x = 0.25) +
  # geom_tippoint(aes(shape = tree, colour = sample), size = 3, position = position_nudge(x = 0.1)) +
  # geom_text(aes(label = tree$node.label), hjust = 1, vjust = -0.4, size = 3) +
  geom_nodelab(nudge_x = 0.035, size = 2.5) +
  # geom_nodelab(nudge_x = -0.125, nudge_y = 0.55, size = 2) +
  # geom_nodelab(nudge_x = 0.15, nudge_y = 0, size = 2) +
  # theme_tree2() +
  xlim(0,10)

jframe <- data.frame(pos = h[["data"]][["y"]],
                     contig = h[["data"]][["label"]])

jframe <- jframe[order(jframe$pos, decreasing = T), ]
jframe <- jframe[jframe$pos%%1 == 0, ]
jframe <- jframe[-(substr(jframe$contig, 1, 1) %>% grep("[0-9]", .)), ]

# Reorder MSA according to tree
msaforNCBI <- read.fasta("msa/mafft_einsi_superclean/mafft_einsi_2021jcyl_superclean.fasta")
names.reord <- sapply(names(msaforNCBI), function(x) grep(x, jframe$contig)) %>% sort
msaforNCBI <- msaforNCBI[names(names.reord)]
write.fasta(sequences = msaforNCBI, names = names(names.reord), 
            "msa/mafft_einsi_superclean/mafft_einsi_2021jcyl_superclean_reorderNCBI.fasta")

```

### Get ordered viral class

```{r}
sapply(names(names.reord), function(x) meta.frame2[grep(x, meta.frame2$contig), "phyl_taxa"])
```


### Other trees

```{r}
# ## MAFFT tree
# tree <- read.tree("msa/mafft_einsi/mafft/tree_assigned_mafft.nh")
# # tree$tip.label <- substr(tree$tip.label, 5, nchar(tree$tip.label))
# contignames <- gsub("\\.", "_", meta.frame2$contig)
# tree$tip.label <- meta.frame2$short_taxa[sapply(tree$tip.label, function(x) which(x == contignames))]
# 
# ## CLUSTAL tree
# tree <- read.tree("msa/clustalw/phylogenetic_tree.ph")
# tree$tip.label <- meta.frame2$phyl_taxa[sapply(tree$tip.label, function(x) which(x == meta.frame2$contig))]
# 
# ## IQTREE WITH OLD MSA
# tree <- read.tree("../preliminary/iqtree/palmscantrimMAFFT2.fasta.contree")
# tree$tip.label <- meta.frame2$phyl_taxa[sapply(tree$tip.label, function(x) which(x == meta.frame2$contig))]
# 
# ## OLD TREE WITH NEW ASSIGNMENT
# tree <- read.tree("../preliminary/palmscantrimMAFFTDIAMOND_taxa.nwk")
# tree$tip.label <- gsub("\\'", "", tree$tip.label)
# getnodes <- lapply(tree$tip.label, function(x) paste0(str_split(x, "_")[[1]][1:3], collapse = "_")) %>% unlist
# metanodes <- sapply(meta.frame2$contig, function(x) paste0(str_split(x, "_")[[1]][1:3], collapse = "_")) %>% unlist
# tree$tip.label <- meta.frame2$short_taxa[sapply(getnodes, function(x) which(x == metanodes)) %>% unlist]
```

# Genome assembly

https://www.genomedetective.com/app/typingtool/virus/job/705c5112-2bfb-482d-bbbd-767c6e77c13e/

# Tantalus

```{r}
library("tantalus")
library("dbplyr")
library("RPostgres") 
```

```{r}
# Connect to Serratus Database
drv <- DBI::dbDriver("PostgreSQL")
con <- DBI::dbConnect(drv,
                      user="public_reader", 
                      password="serratus",
                      host="serratus-aurora-20210406.cluster-ro-ccz9y6yshbls.us-east-1.rds.amazonaws.com",
                      port=5432, 
                      dbname="summary")
```

